### PR TITLE
Check for changed core files

### LINKED ISSUE #12: Check for changed Core files
Call `Core_Upgrader::check_files()`; for `v1` disallow the switch if `false`.

### PR DESCRIPTION
**Pull request by glagonikas**
Fixes #12

### CODE REVIEW

File: lib/admin-page.php [hunk 1]
```diff
@@ -366,14 +366,44 @@ function classicpress_check_can_migrate() {
...
+	);
+	echo "\n<br>\n";
+	if ( $preflight_checks['wp_core_files_checksums'] ) {
+		_e(
+			'You have not modified any core files.',
+			'switch-to-classicpress'
+		);
+	} else {
+		_e(
+			'It seems you have previously <strong class="cp-emphasis">modified some core files</strong>. You can still migrate to ClassicPress, but be aware that any customisations made previously in those files will be lost.',
```
Comment (nylen) [inline lib/admin-page.php]: > You can still migrate to ClassicPress

This is not currently accurate - this check will block the migration.

I think ideally want a check that _does not_ block the migration.  However, this isn't something that exists yet, so we probably need a new icon type for this case.

File: lib/admin-page.php [hunk 2]
```diff
@@ -366,14 +366,44 @@ function classicpress_check_can_migrate() {
...
 	// TODO: Any other checks needed?
 
 	echo "</table>\n";
 
 	if (
 		$preflight_checks['wp_version'] &&
 		$preflight_checks['php_version'] &&
-		$preflight_checks['wp_http_supports_ssl']
+		$preflight_checks['wp_http_supports_ssl'] &&
+		$preflight_checks['wp_core_files_checksums']
```
Comment (nylen) [inline lib/admin-page.php]: If this check is blocking the migration, then `$preflight_checks['wp_core_files_checksums']` also needs to be verified in `lib/update.php`.

If this check is not going to block the migration, then we need to remove this conditional here.

Comment (glagonikas) [reply lib/admin-page.php]: decided to make the checksums non-blocking and added a warn level

**Comment by nylen** [comment]
As a follow-up step in a later PR, perhaps we can show a list of which core files were modified.  However, if no one ever sees this check, it may not be worth it.

**Comment by glagonikas** [comment]
was thinking about that, but couldn't find a built-in function that does that

**Comment by nylen** [comment]
I went ahead and implemented showing the list of modified files here.

We needed to use our own function anyway, because it wasn't possible to tell the difference between "there was an error fetching checksums" (which happens for development versions of WP) or "there were modified files".

Here are the possible states of this new check now:

### No files modified

2018-10-25t16 53 27-05 00

### Files modified

2018-10-25t16 56 16-05 00

### Unknown (development version)

2018-10-25t16 53 08-05 00

**Pull request merged**